ddp-workspace/projects/ddp-dsm-ui/src/app/scan/scan.component.ts

```diff
@@ -450,4 +455,49 @@ export class ScanComponent implements OnInit {
     }
     return null;
   }
+
+  public areScanPairsInvalid(): boolean {
+    if(!this.initialScan)
+    {
+      return this.scanPairs.length < 2;
+    }
+
+    //No valid pair has been made yet, so we want a disabled button
+    if(this.scanPairs.length === 1) {
+      return true;
+    }
+    for (const {leftValue, rightValue} of this.scanPairsValue) {
+      //If both are null I want the button to be enabled, so return false
+      if(rightValue === '' && leftValue === '') {
```

Comment:
Here you are checking whether the right and left values equal an empty string or not. I think the comment is misleading - we don't check for `null`.

P.S. The button is not enabled if both are empty strings

Reply:
I fixed the comment to be correct I think. The button should disabled when both are empty and there is only one pair. The button is enabled if the most recent pair is empty but all the previous ones are not.